.gitignore

```diff
@@ -96,6 +96,17 @@ Temporary Items
 .spack-env/
 spack.lock
 
+#########
+# CMake #
+#########
+CMakeUserPresets.json
+
+# Directories which are created and filled by FetchContent
+share/openPMD/thirdParty/json/
+share/openPMD/thirdParty/toml11/
+share/openPMD/thirdParty/catch2/
+share/openPMD/thirdParty/pybind11/
```

Comment:
We should be aware that with this, the build folder will modify the source tree.
What does the workflow look like switching between different openPMD versions that specify different dependency versions. Will FetchContent notice and update the dependencies' versions upon configuring?

Reply:
That is true and might be a problem. I don't have experience with this.

Moving from lower to higher versions of OpenPMD should not be a problem. These directories have been deleted using `git rm` and should therefore be deleted when moving forward. FetchContent can then download into these directories w/o problem.

A problem might arise if one moves from a higher to a lower version. If the directories have been populated by FetchContent and git tries to add the old libraries again in the same directories I can not predict the behavior.

It would be possible to use a different directory with FetchContent until we are sure that it works as expected.

Reply:
I'll move this to the build directory, as we do in other project.